torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1470,6 +1499,188 @@ def _make_tensor_helper(shape, low=None, high=None):
 
     return samples
 
+foreach_unary_op_db: List[OpInfo] = [
+    ForeachUnaryFuncInfo('_foreach_neg',
+                         method=torch._foreach_neg,
+                         inplace=torch._foreach_neg_,
+                         ref=torch.neg,
+                         dtypes=all_types_and_complex(),
+                         dtypesIfCPU=all_types_and_complex(),
+                         dtypesIfCUDA=all_types_and_complex(),
+                         sample_inputs_func=sample_inputs_foreach,
+                         safe_casts_outputs=False),
+
+    ForeachUnaryFuncInfo('_foreach_sqrt',
+                         method=torch._foreach_sqrt,
+                         inplace=torch._foreach_sqrt_,
+                         ref=torch.sqrt,
+                         dtypes=floating_types(),
+                         dtypesIfCPU=floating_and_complex_types_and(torch.bfloat16),
+                         dtypesIfCUDA=floating_and_complex_types_and(torch.half)),
+
+    ForeachUnaryFuncInfo('_foreach_exp',
```

Comment:
These names could probably also be auto-generated by default from a string like 'exp', and the reference op also acquired from the same string automatically.

Reply:
Done